gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -618,6 +618,10 @@ private JobStatus pollJobStatus(DagNode<JobExecutionPlan> dagNode) {
     synchronized Map<String, Set<DagNode<JobExecutionPlan>>> submitNext(String dagId) throws IOException {
       Dag<JobExecutionPlan> dag = this.dags.get(dagId);
       Set<DagNode<JobExecutionPlan>> nextNodes = DagManagerUtils.getNext(dag);
+
+      // Set flow status to running if it isn't already
+      DagManagerUtils.emitFlowEvent(this.eventSubmitter, dag, TimingEvent.FlowTimings.FLOW_RUNNING);
```

Comment:
initialize(Dag) should be a better place to emit this event?

Reply:
Good point, moved it there